src/Blocks/sources.jl

```diff
@@ -1,3 +1,57 @@
+# Define and register smooth functions
+# These are "smooth" aka differentiable and avoid Gibbs effect
+# These follow: `offset` + `smooth_wave` * `smooth_step` with zero output for `t < start_time`
+function smooth_cos(x, δ, f, amplitude, ϕ, offset, start_time)
+    offset + amplitude * cos(2*π*f*(x - start_time) + ϕ) * smooth_step(x, δ, one(x), zero(x), start_time)
+end
+
+function smooth_damped_sin(x, δ, f, amplitude, damping, ϕ, offset, start_time)
+    offset + exp((start_time - x)*damping)*amplitude*sin(2*π*f*(x - start_time) + ϕ) * smooth_step(x, δ, one(x), zero(x), start_time)
+end
+
+function smooth_ramp(x, δ, height, duration, offset, start_time)
+    offset + height/(duration) * (smooth_xH(x, δ, start_time) - smooth_xH(x, δ, start_time+duration))
+end
+
+function smooth_sin(x, δ, f, amplitude, ϕ, offset, start_time)
+    offset + amplitude * sin(2*pi*f*(x - start_time) + ϕ) * smooth_step(x, δ, one(x), zero(x), start_time)
+end
+
+function smooth_square(x, δ, f, amplitude, offset, start_time)
+    offset + amplitude*2atan(sin(2π*(x - start_time)*f)/δ)/π * smooth_step(x, δ, one(x), zero(x), start_time)
+end
+
+function smooth_step(x, δ, height, offset, start_time)
+    offset + height*(atan((x - start_time)/δ)/π + 0.5)
+end
+
+function smooth_triangular(x, δ, f, amplitude, offset, start_time)
+    offset + amplitude * (1-2acos((1 - δ)sin(2π*(x - start_time)*f))/π) * smooth_step(x, δ, one(x), zero(x), start_time)
+end
+
+function smooth_xH(x, δ, tₒ)
+    0.5*(x-tₒ) * (1+((x-tₒ)/sqrt((x-tₒ)^2+δ^2)))
+end
+
+function square(x, f, amplitude, offset, start_time)
+    offset + (x > start_time) * (amplitude * (4*floor(f*(x - start_time)) - 2*floor(2*(x - start_time)*f) + 1))
+end
+
+function triangular(x, f, amplitude, offset, start_time)
+    p = 1/f # period
+    offset + (x > start_time) * (4 * amplitude * f * abs(abs((x - p/4 - start_time) % p) - p/2) - amplitude)
+end
+
+@register_symbolic smooth_cos(x, δ, f, amplitude, ϕ, offset, start_time)
+@register_symbolic smooth_damped_sin(x, δ, f, amplitude, damping, ϕ, offset, start_time)
+@register_symbolic smooth_ramp(x, δ, height, duration, offset, start_time)
+@register_symbolic smooth_sin(x, δ, f, amplitude, ϕ, offset, start_time)
+@register_symbolic smooth_square(x, δ, f, amplitude, offset, start_time)
+@register_symbolic smooth_step(x, δ, height, offset, start_time)
+@register_symbolic smooth_triangular(x, δ, f, amplitude, offset, start_time)
+@register_symbolic triangular(x, f, amplitude, offset, start_time)
+@register_symbolic square(x, f, amplitude, offset, start_time)
```

Comment:
Why register these?

Reply:
Earlier we didn't assign parameters while declaring them inside components. So `@register` was necessary.

But yeah, now its no longer required. Removed